PayPal/src/main/java/com/braintreepayments/api/PayPalAccount.java

```diff
@@ -36,6 +63,16 @@ void setClientMetadataId(String clientMetadataId) {
         this.clientMetadataId = clientMetadataId;
     }
 
+    /**
+     * @param shouldValidate Flag to denote when the associated {@link PaymentMethodNonce}
+     *                       will be validated. When set to {@code true}, the {@link PaymentMethodNonce}
+     *                       will be validated immediately. When {@code false}, the {@link PaymentMethodNonce}
+     *                       will be validated when used by a server side library for a Braintree gateway action.
+     */
+    public void setShouldValidate(boolean shouldValidate) {
+        this.shouldValidate = shouldValidate;
+    }
```

Comment:
Maybe we could remove these for all classes except `Card`?

We kept `shouldValidate` for non Credit Card types, however, we aren't sure if these are needed. In iOS, the `shouldValidate` property is only defined for `BTCard` and it's omitted for the other payment method types.

Reply:
It is being set to `false` for PayPalCheckoutRequests internally to prevent vaulting. This may be another solution to prevent the merchant from having to call `shouldValidate`.

Reply:
We went ahead and removed these `setShouldValidate` methods and aligned the behavior with iOS. @demerino does this seem correct?